docs/build/js-client/10-marine-js.md

```diff
@@ -2,11 +2,36 @@
 
 > This section is experimental as the API is not stable and probably will change.
 
+## What is Marine JS
+
+### Intro
+When JS client starts up, it starts with initialization Marine JS runtime.
+The runtime hosts wasm services and even allows you to register your own wasm services.
+For example, AquaVM service for processing particles and Air code resides in Marine JS.
+That's why Marine JS is a foundation of JS client. 
+
+### Use cases
```

Comment:
not sure those are really uses cases

Reply:
why?

Reply:
fundamentally, running a js peer requires a developer to self host, which is a serverless anti-pattern, and such peers don't participate in supporting the Fluence network commons, e..g, offer routing or relay services. so the question is what use cases exist that warrant self-hosting a peer, which in fact actually acts like a service rather than a peer.
you listed three cases, which i don't find all that convincing. 
> Heavy computations, CPU intensive tasks
the marketplace offers pretty powerful servers at a fractional cost. however, i can see two related cases that may justify running a peer (actually should be multiple with failover management):
* long running compute jobs, such as training AI models, that exceed the maximum Fluence serverless duration
* compute jobs requiring GPUs

> Would prefer to write service in Rust or other language with compilation to WASM
i don't understand how that supports the use of a self-hosted JS solution. all the services deployed to the Fluence network are currently written in Rust and compiled to wasm-wasi, no JS needed.

> Cross-platform service which works both on Nox peer and JS client peer
also don't quite get the proposition. one of the major points of Wasm is cross-platform and i don't see how the js peer is of any help. unless, you have a effector dependency you can capture with mounted binaries ... maybe because no capacity provider is offering the needed binary, which maybe another use case.
for all the use cases, though, the underlying value arises from self-hosting which in and by itself does not inherently favor the js peer over (self-hosting) a Nox instance, does it?